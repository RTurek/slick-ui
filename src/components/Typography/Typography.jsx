import React from 'react';
import MaterialTypography from '@material-ui/core/Typography';
import { useTheme } from '@material-ui/core/styles';
import { PropTypes } from 'prop-types';

const Typography = (props) => {
  const theme = useTheme();
  const style = {
    ...props.style
  };
  switch (props.color) {
    case 'success':
    case 'alert':
    case 'urgent':
    case 'warning':
    case 'lightGrey':
    case 'mediumGrey':
    case 'modernBlack': {
      const { color, ...otherProps } = props;
      style.color = theme.palette[color].main;
      return <MaterialTypography {...otherProps} style={style} />;
    }
    case 'contrastPrimary': {
      const { color, ...otherProps } = props;
      style.color = theme.palette.primary.contrastText;
      return <MaterialTypography {...otherProps} style={style} />;
    }
    case 'contrastSecondary': {
      const { color, ...otherProps } = props;
      style.color = theme.palette.secondary.contrastText;
      return <MaterialTypography {...otherProps} style={style} />;
    }
    default:
  }
  return <MaterialTypography {...props} />;
};

Typography.propTypes = {
  color: PropTypes.string,
  style: PropTypes.shape({}),
  theme: PropTypes.shape({
    palette: PropTypes.shape({
      primary: PropTypes.shape({
        contrastText: PropTypes.string
      }),
      secondary: PropTypes.shape({
        contrastText: PropTypes.string
      })
    })
  })
};

Typography.defaultProps = {
  color: 'textPrimary',
  style: undefined,
  theme: {
    palette: {
      primary: {
        contrastText: 'white'
      },
      secondary: {
        contrastText: 'white'
      }
    }
  }
};

export default Typography;
