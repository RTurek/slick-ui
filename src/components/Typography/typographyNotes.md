#### Typography
---

This is a wrapper component for Material Typography

#### Notes
---


#### Sources
---
- https://mui.com/components/typography/
