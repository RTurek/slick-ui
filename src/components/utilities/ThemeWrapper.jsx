import React from 'react';
import PropTypes from 'prop-types';
import { createTheme } from '@material-ui/core/styles';
import { ThemeProvider } from '@material-ui/styles';
import { COLORS } from '@styles/colors';

class ThemeWrapper extends React.Component {
  constructor(props) {
    super(props);

    const primaryThemeColor = COLORS.primary
    const secondaryThemeColor = COLORS.secondary
    const textPrimaryThemeColor = COLORS.textPrimary;
    const textSecondaryThemeColor = COLORS.textSecondary;
    const backgroundThemeColor = COLORS.background;
    const alertThemeColor = COLORS.alert;
    const urgentThemeColor = COLORS.urgent;
    const successThemeColor = COLORS.success;
    const warningThemeColor = COLORS.warning;
    const crGreenThemeColor = COLORS.crGreen;
    const rpOrangeThemeColor = COLORS.rpOrange;
    const darkGreyThemeColor = COLORS.darkGrey;
    const mediumGreyThemeColor = COLORS.mediumGrey;
    const lightGreyThemeColor = COLORS.lightGrey;
    const modernBlackThemeColor = COLORS.modernBlack;

    this.muiTheme = createTheme({
      typography: {
        // fontFamily: primaryFontFamily,
        useNextVariants: true,
        h1: {
          letterSpacing: '-1.5px'
        },
        h2: {
          letterSpacing: '-0.5px'
        },
        h3: {
          lineHeight: 1,
          letterSpacing: '0px',
          fontWeight: '300'
        },
        h4: {
          fontSize: '1.875rem',
          letterSpacing: '0.25px',
          lineHeight: 1.0667
        },
        h5: {
          letterSpacing: '0px',
          lineHeight: 1.0833
        },
        h6: {
          letterSpacing: '0.15px',
          lineHeight: 1.1
        },
        subtitle1: {
          letterSpacing: '0.15px',
          lineHeight: 1.125
        },
        subtitle2: {
          letterSpacing: '0.1px',
          lineHeight: 1.1428
        },
        body1: {
          letterSpacing: '0.25px',
          lineHeight: 1.25
        },
        body2: {
          letterSpacing: '0.15px',
          lineHeight: 1.28571
        },
        body1Next: {
          letterSpacing: '0.25px',
          lineHeight: 1.25
        },
        body2Next: {
          letterSpacing: '0.15px',
          lineHeight: 1.28571
        },
        button: {
          fontWeight: 700,
          letterSpacing: '1.25px'
        },
        buttonNext: {
          fontWeight: 700,
          letterSpacing: '1.25px'
        },
        caption: {
          letterSpacing: '0.2px',
          lineHeight: 1.33333
        },
        overline: {
          fontWeight: 600,
          fontSize: '0.625rem',
          letterSpacing: '1.5px',
          lineHeight: 1.4
        },
        fontWeightLight: 300,
        fontWeightMedium: 600,
        fontWeightRegular: 400
      },
      palette: {
        primary: {
          main: primaryThemeColor
        },
        secondary: {
          main: secondaryThemeColor,
          contrastText: '#FFF'
        },
        error: {
          main: alertThemeColor
        },
        text: {
          primary: textPrimaryThemeColor,
          secondary: textSecondaryThemeColor
        },
        alert: {
          main: alertThemeColor
        },
        urgent: {
          main: urgentThemeColor
        },
        success: {
          main: successThemeColor
        },
        warning: {
          main: warningThemeColor
        },
        backgroundColor: {
          main: backgroundThemeColor
        },
        mediumGrey: {
          main: mediumGreyThemeColor
        },
        lightGrey: {
          main: lightGreyThemeColor
        },
        crGreen: {
          main: crGreenThemeColor
        },
        darkGrey: {
          main: darkGreyThemeColor
        },
        rpOrange: {
          main: rpOrangeThemeColor
        },
        modernBlack: {
          main: modernBlackThemeColor
        }
      },
      overrides: {
        MuiTypography: {
          root: {
            display: 'block'
          }
        }
      },
      props: {
        MuiButtonBase: {
          TouchRippleProps: {
            style: {
              color: lightGreyThemeColor
            }
          }
        }
      }
    });
  }

  render() {
    return (
      <ThemeProvider theme={this.muiTheme}>
        {this.props.children}
      </ThemeProvider>
    );
  }
}

ThemeWrapper.propTypes = {
  children: PropTypes.node
};

ThemeWrapper.defaultProps = {
  children: null
};

export default ThemeWrapper;
