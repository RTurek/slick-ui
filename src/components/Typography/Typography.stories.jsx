import React from 'react';
import { withInfo } from '@storybook/addon-info';
import { boolean, select, object, text } from '@storybook/addon-knobs';
import { storybookStyles } from '../../../.storybook/storybookTheming';
import StoryWrapper from '../utilities/StoryWrapper';
import Typography from './Typography';
import typographyNotes from './typographyNotes.md';

const VARIANTS = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'subtitle1',
  'subtitle2',
  'body1',
  'body2',
  'caption',
  'button',
  'overline',
  'srOnly',
  'inherit'
];

export default {
  title: 'Core / Typography',

  decorators: [
    withInfo({
      text: typographyNotes,
      header: true,
      inline: true,
      source: false,
      styles: storybookStyles,
      propTables: [Typography],
      propTablesExclude: [],
      excludedPropTypes: ['theme']
    })
  ]
};

export const typographyList = () => (
  <StoryWrapper>
    <div style={{ backgroundColor: '#fff' }}>
      {VARIANTS.map(variant => (
        <div style={{ margin: '10px' }} key={variant}>
          <Typography
            variant={variant}
            {...{
              align: select('align', [
                'initial',
                'inherit',
                'left',
                'center',
                'right',
                'justify'
              ], 'left'),
              color: select('color', [
                'inherit',
                'primary',
                'secondary',
                'textPrimary',
                'textSecondary',
                'error',
                'warning',
                'success'
              ]),
              display: select('display', ['initial', 'block', 'inline']),
              gutterBottom: boolean('gutterBottom', false),
              noWrap: boolean('noWrap', false),
              paragraph: boolean('paragraph', false),
              style: object('style', { display: 'inherit' })
            }}
          >
            {variant} {text('text', 'Slick-UI')}
          </Typography>
        </div>
      ))}
    </div>
  </StoryWrapper>
);

typographyList.story = {
  name: 'Typography List'
};
