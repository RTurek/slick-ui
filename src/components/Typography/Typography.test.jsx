import React from 'react';
import renderer from 'react-test-renderer';
import Typography from './Typography';
import ThemeWrapper from '../../styles/ThemeWrapper';

test('Typography should render', () => {
  const component = (
    <ThemeWrapper>
      <Typography
        variant="body1"
        color="primary"
      >
        This Is A Test
      </Typography>
    </ThemeWrapper>
  );
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
