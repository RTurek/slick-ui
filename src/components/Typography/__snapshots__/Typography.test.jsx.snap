// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
  theme={
    Object {
      "palette": Object {
        "primary": Object {
          "contrastText": "white",
        },
        "secondary": Object {
          "contrastText": "white",
        },
      },
    }
  }
>
  This Is A Test
</p>
`;
